js/client-chat.js

```diff
@@ -567,11 +567,20 @@
 					for (var i=0, len=targets.length; i<len; i++) {
 						targets[i] = targets[i].trim();
 					}
-					switch (targets[0]) {
+					switch (targets.shift()) {
 					case 'add':
-						for (var i=1, len=targets.length; i<len; i++) {
-							highlights.push(targets[i].trim());
-						}
+						var test;
+						for (var i=0, len=targets.length; i<len; i++) {
+							if (/[\\^$*+?()|{}[\]]/.test(targets[i])) {
+								// Catch any errors thrown by newly added regular expressions so they don't break the entire highlight list
+								try {
+									test = new RegExp(targets[i]);
+								} catch (e) {
+									return this.add(e.message.substr(0, 28) === 'Invalid regular expression: ' ? e.message : 'Invalid regular expression: /' + targets[i] + '/: ' + e.message);
```

Comment:
The error message on browsers other than Chrome/Chromium don't include "Invalid regular expression: " or the regex in which the error occurs in the error message, so this line adds those to keep the error message consistent across all browsers